subs/pantry/src/Pantry.hs

```diff
@@ -1006,6 +1010,27 @@ runPantryApp f = runSimpleApp $ do
           }
         f
 
+-- | Like 'runPantryApp', but you can give the custom HPack executable to it.
+--
+-- @since 0.1.0.0
+runPantryAppWithCustomHpack :: MonadIO m => Path Abs File -> RIO PantryApp a -> m a
+runPantryAppWithCustomHpack fp f = runSimpleApp $ do
```

Comment:
It seems strange to have a call to `runSimpleApp` here. Why not have it live in `RIO`? Concrete concern: this will disrupt any kind of logging already set up in the program.

Reply:
I have removed this function altogether. This function was added for it's use in one of the unit testing functions. I have now exported a lens function from this module for achieving the same.